host-controller/src/main/java/org/jboss/as/host/controller/mgmt/HostControllerRegistrationHandler.java

```diff
@@ -204,6 +205,22 @@ public HostControllerRegistrationHandler(ManagementChannelHandler handler, Domai
          */
         ModelNode executeReadOnly(ModelNode operation, Resource model, OperationStepHandler handler, ModelController.OperationTransactionControl control);
 
+        /**
+         * Registers a slave host controller. A non-exclusive read lock is held for the duration of this registration
+         * process, blocking writes, but allowing the initiation of additional read locks.
+         * @param operationId - the operation ID for this registration
+         * @param hostInfo - the registering host's HostInfo
+         * @throws OperationFailedException - if the lock acquisition fails
+         */
+        void beginRegistration(final Integer operationId, final HostInfo hostInfo) throws OperationFailedException;
```

Comment:
Should these methods be renamed and the javadoc changed? For example, beginRegistration doesn't "Register a slave host controller". It just acquires a lock.

A generic name like beginRegistration is ok if you want flexibility to do more stuff in the future. In that case the javadoc just needs a tweak so it doesn't sound like it does more that it really does.